### PR TITLE
add enableAntiAlias for CCView fireball/issues/2836

### PR DESCRIPTION
**Pull request by knoxHuang**
Re: cocos-creator/fireball#2836

Changes proposed in this pull request:
- 提供关闭图像平滑渲染的方法

> Makes sure these boxes are checked before submitting your PR - thank you!
> - [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
> - [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
> - To official teams:
>   - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
>   - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/platform/CCView.js [hunk 1]
```diff
@@ -416,6 +419,53 @@ var View = cc._Class.extend({
...
+                    }
+                }
+            }
+        }
+        else if(cc._renderType === cc.game.RENDER_TYPE_CANVAS) {
+            var ctx = cc._canvas.getContext('2d');
+            ctx.imageSmoothingEnabled = enabled;
+            ctx.mozImageSmoothingEnabled = enabled;
+            // refresh canvas
+            var dirtyRegion = cc.rendererCanvas._dirtyRegion;
```
Comment (jareguo) [inline cocos2d/core/platform/CCView.js]: 这里有点 hack 吧，没有现成方法吗

Comment (knoxHuang) [inline cocos2d/core/platform/CCView.js]: @dabingnn 看你咯，j 神说这里很 hack

Comment (pandamicro) [inline cocos2d/core/platform/CCView.js]: 还有一种办法是先关闭脏矩形，然后用一个 scheduleOnce 回调在回调中重新开启，不过我觉得添加 dirtyRegion 这种办法相对还好一些

Comment (dabingnn) [inline cocos2d/core/platform/CCView.js]: 我不太清楚当横竖屏切换的时候，咱们怎么处理的，这一块应该可以用和横竖屏切换差不多的逻辑

**Comment by knoxHuang** [comment]
传送门： https://github.com/cocos-creator/engine/pull/998

**Pull request closed without merging**